### ISSUE TITLE
Can MXNet support locally-connected layer?

### ISSUE THREAD

**Issue by yuzhms** [apache/incubator-mxnet#15570]
Can MXNet support locally-connected layer, which is conv layer without share weight, like [`tf.keras.layers.LocallyConnected2D`](https://www.tensorflow.org/api_docs/python/tf/keras/layers/LocallyConnected2D).

**Comment by vrakesh** [comment]
@yuzhms Thanks for the question, can you provide more details on requested feature support , with an example . That would explain the need for it

@mxnet-label-bot [Feature request]

**Comment by yuzhms** [comment]
It's quite simple.

 If we have an input with shape `(BxC_inxHxW)`，we use a convolution kernel with stride =3, and we want to output feature maps with `C_out` channels, the convolution layer will create weights with shape `(C_out, C_in, 3, 3)`. So the operation in each space position share weight.

Locally connected layer operate as same as the convolution layer, except in each space position, it use different weight. So the weights now have shape of `(C_out, C_in, 3, 3, H_out, W_out)`.

see[this link](https://prateekvjoshi.com/2016/04/12/understanding-locally-connected-layers-in-convolutional-neural-networks/) for more comprehensive explanation.
